packages/contracts/test/OwnershipTest.js

```diff
@@ -77,38 +77,59 @@ contract('All Liquity functions with onlyOwner modifier', async accounts => {
 
   describe('TroveManager', async accounts => {
     it("setAddresses(): reverts when called by non-owner, with wrong addresses, or twice", async () => {
-      await testSetAddresses(troveManager, 14)
+      const dumbContract = await GasPool.new()
+      const params = Array(15).fill(dumbContract.address)
+
+      // Attempt call from alice
+      await th.assertRevert(troveManager.setAddresses(params, { from: alice }))
+
+      // Attempt to use zero address
+      for (let i = 0; i < params.length; i++) {
+        const newParams = [...params]
+        newParams[i] = th.ZERO_ADDRESS
+        await th.assertRevert(troveManager.setAddresses(newParams, { from: owner }),  'Account cannot be zero address')
+      }
+      // Attempt to use non contract
+      for (let i = 0; i < params.length; i++) {
+        const newParams = [...params]
+        newParams[i] = bob
+        await th.assertRevert(troveManager.setAddresses(newParams, { from: owner }),  'Account cannot be zero address')
+      }
+
+      // Owner can successfully set any address
+      const txOwner = await troveManager.setAddresses(params, { from: owner })
+      assert.isTrue(txOwner.receipt.status)
```

Comment:
what's this?

Reply:
This test has a function that verify that if any parameter is not a contract or an address(0) reverts the setAddresses function, but troveManager receives an array with all the addresses now , so , we have to test the same in another way

Reply:
This test has a function that verify that if any parameter is not a contract or an address(0) reverts the setAddresses function, but troveManager receives an array with all the addresses now , so , we have to test the same in another way